packages/api/src/services/gateways/dataStoreGateways/postgres/cases/filters/exceptionsAndTriggers.ts

```diff
@@ -0,0 +1,57 @@
+import type { User } from "@moj-bichard7/common/types/User"
+import type postgres from "postgres"
+import type { Row } from "postgres"
+
+import Permission from "@moj-bichard7/common/types/Permission"
+import { userAccess } from "@moj-bichard7/common/utils/userPermissions"
+
+import type { Filters } from "../../../../../../types/CaseIndexQuerystring"
+
+import {
+  resolutionStatusCodeByText,
+  ResolutionStatusNumber
+} from "../../../../../../useCases/dto/convertResolutionStatus"
+
+const getExcludedTriggers = (excludedTriggers?: string[]): string[] =>
+  excludedTriggers && excludedTriggers.length > 0 ? excludedTriggers : []
+
+export const exceptionsAndTriggers = (
+  sql: postgres.Sql,
+  user: User,
+  filters: Filters
+): postgres.PendingQuery<Row[]> => {
+  const resolutionStatus = filters.caseState
+    ? (resolutionStatusCodeByText(filters.caseState) ?? ResolutionStatusNumber.Unresolved)
+    : ResolutionStatusNumber.Unresolved
+  let exceptionsSql
+  let triggersSql
+
+  if (userAccess(user)[Permission.Exceptions]) {
+    exceptionsSql = sql`(el.error_count > 0 AND el.error_status = ${resolutionStatus})`
+  }
+
+  if (userAccess(user)[Permission.Triggers]) {
+    const excludedTriggers = getExcludedTriggers(user.excluded_triggers?.split(","))
```

Comment:
Nitpick: Since `getExcludedTriggers` is just a local function, wondering whether we could move the `split` call into there so this is a bit cleaner and ends up simply being `getExcludedTriggers(user.excluded_triggers)`?